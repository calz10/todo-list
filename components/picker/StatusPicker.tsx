import React from 'react'
import { Picker } from '@react-native-picker/picker';
import { TodoActionTypes } from '@/redux/todos';

type OptionITem <TItem>= {
  label: string
  value: TItem
}

type PickerProps<T>= {
  status?: OptionITem<T>['value']
  options: Array<OptionITem<T>>
  onChange: (val: T) => void
}

export function StatusPicker<T>({
  onChange,
  status,
  options
}: PickerProps<T>) {
  return (
    <Picker
      selectedValue={status}
      mode='dialog'
      onValueChange={onChange}>
        {
         options.map(({ label, value }) => (
            <Picker.Item key={label} label={label} value={value} />
          ))
        }
    </Picker>
  )
}