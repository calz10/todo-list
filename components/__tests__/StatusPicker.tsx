import * as React from 'react';
import renderer from 'react-test-renderer';

import { StatusPicker } from '../picker/StatusPicker';

it(`renders correctly`, () => {
  const tree = renderer.create(<StatusPicker
    onChange={() => {}}
    status={1}
    options={[
      {
      label: 'All',
      value: 1
    },
      {
      label: 'Pending',
      value: 2
    },
      {
      label: 'Complete',
      value: 3
    },
  ]}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
