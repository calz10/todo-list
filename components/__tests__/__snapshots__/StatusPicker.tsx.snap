// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <RNCPicker
    items={
      [
        {
          "label": "All",
          "testID": undefined,
          "textColor": undefined,
          "value": "1",
        },
        {
          "label": "Pending",
          "testID": undefined,
          "textColor": undefined,
          "value": "2",
        },
        {
          "label": "Complete",
          "testID": undefined,
          "textColor": undefined,
          "value": "3",
        },
      ]
    }
    numberOfLines={1}
    onChange={[Function]}
    selectedIndex={0}
    style={
      [
        {
          "height": 216,
        },
        undefined,
      ]
    }
  />
</View>
`;
